lib/persist.php

```diff
@@ -86,7 +89,11 @@ function sanitizeChartConfig($charts) {
 if (inputIsValid($data)) {
   try {
     sanitizeChartConfig($data->charts);
-    $module->setProjectSetting('chart-definitions', $data->charts);
+    if ($module->canEditVizrCharts()) {
+      $module->setProjectSetting('chart-definitions', $data->charts);
+    } else {
+      $response->errors = array("You do not have permission to modify chart definitions.");
+    }
     $response->item_count = 1;
```

Comment:
Setting the item count should be moved under `$module->setProjectSetting(...);` for consistency.